src/Core/Domain/Product/SpecificPrice/Command/AddProductSpecificPriceCommand.php

```diff
@@ -322,19 +323,19 @@ public function setCurrencyId(int $currencyId): self
     }
 
     /**
-     * @return int
+     * @return CountryIdInterface
      */
-    public function getCountryId(): int
+    public function getCountryId(): CountryIdInterface
     {
         return $this->countryId;
     }
 
     /**
-     * @param int $countryId
+     * @param CountryIdInterface $countryId
      *
      * @return $this
      */
-    public function setCountryId(int $countryId): self
+    public function setCountryId(CountryIdInterface $countryId): self
```

Comment:
Same here,  you shouldn't change the setter's prototype

Reply:
copy that too !